### PR TITLE
Add the mangled_args property to kernel_api types.

### LINKED ISSUE #1242: Breakpoint with function name not working for numba-dpex.
Breakpoint by passing function name as argument doesn't work for numba-dpex.
gdb version: GNU gdb (Intel(R) Distribution for GDB* 2024.0.0) 13.1
dpcpp version: Intel(R) oneAPI DPC++/C++ Compiler 2024.0.0 (2024.0.0.20231017)
**Actual:**
```
$  ONEAPI_DEVICE_SELECTOR=opencl:cpu NUMBA_OPT=0 NUMBA_DPEX_OPT=0 gdb-oneapi -q python
(gdb) b side-by-side.py:common_loop_body
(gdb) r side-by-side.py --api=numba-ndpx-kernel
...
[0. 1. 2. 3. 4. 5. 6. 7. 8. 9.]
[0. 1. 2. 3. 4. 5. 6. 7. 8. 9.]
[10.  11.5 13.  14.5 16.  17.5 19.  20.5 22.  23.5]
Done...
```
This breakpoint by function name `common_loop_body` doesn't get triggered for numba-dpex but works fine for numba.
**Expected:**
```
$  ONEAPI_DEVICE_SELECTOR=opencl:cpu NUMBA_OPT=0 NUMBA_DPEX_OPT=0 gdb-oneapi -q python
(gdb) b side-by-side.py:common_loop_body 
(gdb) r side-by-side.py --api=numba
Thread 1 "python" hit Breakpoint 1.2, __main__::common_loop_body[abi:v2][abi:c8tJTC_2fWwGaMLShh6CjAIxrKR0t9qKIYaQ0qDQwmMEsTAA_3d_3d](float, float) (param_a=0, param_b=0) at side-by-side.py:16
16          param_c = param_a + 10  # Set breakpoint here
...

### PR DESCRIPTION
**Pull request by diptorupd**
- [X] Have you provided a meaningful PR description?
Numba-dpex's custom types such as `ItemType` and `NdItemType` are currently not properly demangled and makes the demangled function signature hard to read. The PR fixes the issue by adding the `mangled_args` property to these types.

In addition, the `SpirvKernelTarget` was updated to use the `numba_dpex.core.utils.itanium_mangler`.

Before (look at the first argument "Item"):

```bash
Thread 2.129 hit Breakpoint 1, with SIMD lanes [0-1], __main__::data_parallel_sum_242(
    Item_3c1_3e, 
    DpnpNdArray<float, 1, C, level_zero_gpu>,
    DpnpNdArray<float, 1, C, level_zero_gpu>,
    DpnpNdArray<float, 1, C, level_zero_gpu>
) (item=..., a=..., b=..., c=...) at simple_sum.py:13
```

After:

```bash
Thread 2.129 hit Breakpoint 1, with SIMD lanes [0-1], __main__::data_parallel_sum_242(
    ItemType<1>,
    DpnpNdArray<float, 1, C, level_zero_gpu>,
    DpnpNdArray<float, 1, C, level_zero_gpu>,
    DpnpNdArray<float, 1, C, level_zero_gpu>
) (item=..., a=..., b=..., c=...) at simple_sum.py:13
```

- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

Fixes #1242

### CODE REVIEW

**Comment by diptorupd** [comment]
@mpuputti The PR has my latest GDB-related changes I have done based on our triaging of the GDB support on numba-dpex kernels.

The changes do fix setting breakpoints on function names. I will investigate further if there are any unintended side effects.

File: numba_dpex/kernel_api_impl/spirv/target.py [hunk 1]
```diff
@@ -119,6 +120,39 @@ class SPIRVDevice(GPU):
 target_registry[SPIRV_TARGET_NAME] = SPIRVDevice
 
 
+class _DIBuilder(debuginfo.DIBuilder):
```
Comment (ZzEeKkAa) [inline numba_dpex/kernel_api_impl/spirv/target.py:123]: May be something specific to numba dpex is a better choice for naming?
```suggestion
class SPIRVDIBuilder(debuginfo.DIBuilder):
```

**Review by ZzEeKkAa — CHANGES_REQUESTED** [review]
Awesome changes! Thank you!

Finally breakpoints by name are working! Just few nits, otherwise good to go!

File: numba_dpex/kernel_api_impl/spirv/target.py (hunk 1 above)
Comment (diptorupd) [reply numba_dpex/kernel_api_impl/spirv/target.py]: I moved the class into `numba_dpex.core.debuginfo` and renamed it to `DIBuilder`.

File: numba_dpex/tests/test_debuginfo.py [hunk 2]
```diff
@@ -190,7 +190,7 @@ def foo(item, a, b):
     kernel_ir = kcres.library.get_llvm_str()
 
     for tag in ir_tags:
-        assert make_check(kernel_ir, tag)
+        assert not make_check(kernel_ir, tag)
```
[the remark this answers is not in the document]
Comment (diptorupd) [reply numba_dpex/tests/test_debuginfo.py:197]: After these changes the `linkagename` tag is no longer used. So changed the test to a negative test case to make sure `linkagename` is not set. I have added comments make the change clearer.

**Review by ZzEeKkAa — APPROVED** [review]

**Pull request merged**